src/Sulu/Component/Content/Document/Subscriber/ShadowLocaleSubscriber.php

```diff
@@ -204,10 +204,12 @@ public function handlePersistUpdateUrl(PersistEvent $event)
             return;
         }
 
-        $event->getAccessor()->set(
-          'resourceSegment',
-          $shadowLocator
-        );
+        if (!$event->getAccessor()->get('resourceSegment')) {
```

Comment:
we want to set the resource segment to the segment of the shadow locale only if the document has no segment set. if the document already has a segment, we dont want to overwrite it